File: longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) 
    {
        if(nums.length == 0)
        {
            return 0;
        }
        
        Arrays.sort(nums);    
        
        int current_streak = 1;
        int longest_streak = 1;
        
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i] != nums[i-1])
            {
                if(nums[i] == nums[i-1] + 1)
                {
                    current_streak +=1;
                }
                else
                {
                    longest_streak = Math.max(longest_streak,current_streak);
                    current_streak  = 1;
                }
            }
        }
        
        return Math.max(current_streak,longest_streak);
    }
    
}

File: longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) 
    {
        unordered_set<int> s(nums.begin(),nums.end());
        int max_len = 0;
        
        for(auto x : s)
        {
            if(s.find(x-1) != s.end())
            {
                continue;
            }
            
            int len = 1;
            
            while(s.find(x+len) != s.end())
            {
                len++;
            }
            
            max_len = max(max_len,len);
        }
        
        return max_len;
    }
};